### PR TITLE
(ocrvs-2382-fix) Disable back arrow button on landing page

### PR DESCRIPTION
**Pull request by sadmananik**
ocrvs-2382-fix

### CODE REVIEW

File: packages/client/src/components/interface/Header/Header.tsx [hunk 1]
```diff
@@ -288,11 +307,38 @@ class HeaderComp extends React.Component<IProps, IState> {
     this.props.redirectToAuthentication()
   }
 
+  isLandingPage = () => {
+    const role = this.props.userDetails && this.props.userDetails.role
+    const location = this.props.history.location.pathname as any
+    if (
+      FIELD_AGENT_ROLES.includes(role as string) &&
+      [HOME].includes(location)
```
Comment (atiqzaman-dsi) [inline packages/client/src/components/interface/Header/Header.tsx:315]: Why do you need `[HOME]` as an `array`? And why the location is `any`. Can we define a type for that?

File: packages/client/src/components/interface/Header/Header.tsx [hunk 2]
```diff
@@ -288,11 +307,38 @@ class HeaderComp extends React.Component<IProps, IState> {
...
+    ) {
+      return true
+    } else if (
+      NATL_ADMIN_ROLES.includes(role as string) &&
+      [PERFORMANCE_HOME].includes(location)
```
Comment (atiqzaman-dsi) [inline packages/client/src/components/interface/Header/Header.tsx:320]: Same for this line.

File: packages/client/src/components/interface/Header/Header.tsx [hunk 3]
```diff
@@ -288,11 +307,38 @@ class HeaderComp extends React.Component<IProps, IState> {
...
+    ) {
+      return true
+    } else if (
+      SYS_ADMIN_ROLES.includes(role as string) &&
+      [OPERATIONAL_REPORT].includes(location)
+    ) {
+      return true
+    } else if (
+      REGISTRAR_ROLES.includes(role as string) &&
+      [REGISTRAR_HOME].includes(location)
```
Comment (atiqzaman-dsi) [inline packages/client/src/components/interface/Header/Header.tsx:330]: I think you have to extract this to a function and use that for other similar checks

**Review by atiqzaman-dsi — CHANGES_REQUESTED** [review]

File: packages/client/src/components/interface/Header/Header.tsx (hunk 1 above)
Comment (sadmananik) [reply packages/client/src/components/interface/Header/Header.tsx:315]: I used an array to maintain if any role has multiple landing pages. But for the time being, I have changed it to string and also removed any type from the location variable as it automatically infers to string.

File: packages/client/src/components/interface/Header/Header.tsx (hunk 2 above)
Comment (sadmananik) [reply packages/client/src/components/interface/Header/Header.tsx:320]: Have changed it to string.

File: packages/client/src/components/interface/Header/Header.tsx (hunk 3 above)
Comment (sadmananik) [reply packages/client/src/components/interface/Header/Header.tsx:330]: I have returned a single true or false boolean from `isLandingPage` function.

**Review by atiqzaman-dsi — APPROVED** [review]

**Pull request merged**